xarray/namedarray/core.py

```diff
@@ -0,0 +1,435 @@
+        """
+        The imaginary part of the variable.
+
+        See Also
+        --------
+        numpy.ndarray.imag
+        """
+        return self._replace(data=self.data.imag)
+
+    def __dask_tokenize__(self: T_NamedArray):
+        # Use v.data, instead of v._data, in order to cope with the wrappers
+        # around NetCDF and the like
+        from dask.base import normalize_token
+
+        return normalize_token((type(self), self._dims, self.data, self.attrs))
+
+    def __dask_graph__(self: T_NamedArray):
+        return self._data.__dask_graph__() if is_duck_dask_array(self._data) else None
+
+    def __dask_keys__(self: T_NamedArray):
+        return self._data.__dask_keys__()
+
+    def __dask_layers__(self: T_NamedArray):
+        return self._data.__dask_layers__()
+
+    @property
+    def __dask_optimize__(self: T_NamedArray) -> typing.Callable:
+        return self._data.__dask_optimize__
+
+    @property
+    def __dask_scheduler__(self: T_NamedArray) -> typing.Callable:
+        return self._data.__dask_scheduler__
+
+    def __dask_postcompute__(
+        self: T_NamedArray,
+    ) -> tuple[typing.Callable, tuple[typing.Any, ...]]:
+        array_func, array_args = self._data.__dask_postcompute__()
+        return self._dask_finalize, (array_func,) + array_args
+
+    def __dask_postpersist__(
+        self: T_NamedArray,
+    ) -> tuple[typing.Callable, tuple[typing.Any, ...]]:
+        array_func, array_args = self._data.__dask_postpersist__()
+        return self._dask_finalize, (array_func,) + array_args
+
+    def _dask_finalize(
+        self: T_NamedArray, results, array_func, *args, **kwargs
+    ) -> T_NamedArray:
+        data = array_func(results, *args, **kwargs)
+        return type(self)(self._dims, data, attrs=self._attrs)
+
+    @property
+    def chunks(self: T_NamedArray) -> tuple[tuple[int, ...], ...] | None:
+        """
+        Tuple of block lengths for this dataarray's data, in order of dimensions, or None if
+        the underlying data is not a dask array.
+
+        See Also
+        --------
+        Variable.chunk
+        Variable.chunksizes
+        xarray.unify_chunks
+        """
+        return getattr(self._data, "chunks", None)
+
+    @property
+    def chunksizes(
+        self: T_NamedArray,
+    ) -> typing.Mapping[typing.Any, tuple[int, ...]]:
+        """
+        Mapping from dimension names to block lengths for this variable's data, or None if
+        the underlying data is not a dask array.
+        Cannot be modified directly, but can be modified by calling .chunk().
+
+        Differs from variable.chunks because it returns a mapping of dimensions to chunk shapes
+        instead of a tuple of chunk shapes.
+
+        See Also
+        --------
+        Variable.chunk
+        Variable.chunks
+        xarray.unify_chunks
+        """
+        if hasattr(self._data, "chunks"):
+            return Frozen(dict(zip(self.dims, self.data.chunks)))
```

Comment:
you are correct that there is no mutability risk here, because a new dictionary is created each time the property is accessed. i'm going to remove it and see if this causes unexpected behavior elsewhere in the code.